icalevents/icalparser.py

```diff
@@ -241,12 +250,24 @@ def parse_events(content, start=None, end=None, default_span=timedelta(days=7)):
         raise ValueError('Content is invalid!')
 
     calendar = Calendar.from_ical(content)
-    
-    # Find the calendar's timezone info, or use UTC
-    for c in calendar.walk():
-        if c.name == 'VTIMEZONE':
-            cal_tz = gettz(str(c['TZID']))
-            break
+
+    # Keep track of the timezones defined in the calendar
+    timezones = {}
+    for c in calendar.walk('VTIMEZONE'):
+        name = str(c['TZID'])
+        try:
+            timezones[name] = c.to_tz()
+        except IndexError:
+            # This happens if the VTIMEZONE doesn't
+            # contain start/end times for daylight
+            # saving time. Get the system pytz
+            # value from the name as a fallback.
+            timezones[name] = timezone(name)
```

Comment:
Would it be wise to to log a warning here?

Reply:
I don't think it's necessary. If the timezone isn't known locally, the call will fail. If it is known, the only case where it matters is if the rules for daylight saving time have changed.